tools/acrn-crashlog/acrnprobe/conf.rst

```diff
@@ -0,0 +1,379 @@
+
+   digraph {
+      {
+         node [shape=plaintext];
+         "level 1" -> "level 2" -> "level 3";
+      }
+
+      node [shape=box;style="rounded,filled";color=AntiqueWhite;];
+      c1 [ label="crash A\nid 1\ncrash root" ];
+      c2 [ label="crash B\nid 2" ];
+      c3 [ label="crash C\nid 3\ncrash leaf" ];
+      c4 [ label="crash D\nid 4\ncrash leaf" ];
+      { rank = same; "level 1"; c1;}
+      { rank = same; "level 2"; c2; c4;}
+      { rank = same; "level 3"; c3;}
+
+      node [shape=box;style="rounded,dashed";];
+      exp1 [ label="crash B matches fail\nmatch for the next child\nof crash A"];
+      exp2 [ label="crash D matches successfully\nreturn crash D"];
+
+      node [shape=box;style="invis";];
+      "channel" -> c1 [ label="trigger" ]
+      c1 -> {exp1 exp2}
+      exp1 -> c2 -> c3 [ style=dashed dir=none]
+      exp2 -> c4
+   }
+
+Sections
+********
+
+Sender
+======
+
+Example:
+
+.. code-block:: xml
+
+   <sender id="1" enable="true">
+           <name>crashlog</name>
+           <outdir>/var/log/crashlog</outdir>
+           <maxcrashdirs>1000</maxcrashdirs>
+           <maxlines>5000</maxlines>
+           <spacequota>90</spacequota>
+           <uptime>
+                   <name>UPTIME</name>
+                   <frequency>5</frequency>
+                   <eventhours>6</eventhours>
+           </uptime>
+   </sender>
+
+* ``name``:
+  Name of sender. ``Acrnprobe`` uses this label to distinguish different
+  senders.
+  More information about sender, please refer to :ref:`acrnprobe_doc`.
+* ``outdir``:
+  Directory to store generated files of sender. ``Acrnprobe`` will create
+  this directory if it doesn't exist.
+* ``maxcrashdirs``:
+  The maximum serial number of generated ``crash directories``, ``stat directories``
+  and ``vmevent directories``. The serial number will back to 0 if it reaches
+  the maximum. Only used by sender crashlog.
+* ``maxlines``:
+  The maximum lines of file ``history_event``. It will trigger action
+  ``"mv history_event history_event.bak"`` that the lines of ``history_event``
+  reaches the ``maxlines``.
+* ``spacequota``:
+  ``Acrnprobe`` will stop collecting logs if it is true that
+  ``(used space / total space) * 100 > spacequota``. Only used by sender
+  crashlog.
+* ``uptime``:
+  Configuration to trigger ``UPTIME`` event.
+  sub-nodes:
+
+  + ``name``:
+    The name of event.
+  + ``frequency``:
+    Time interval in seconds to trigger ``uptime`` event.
+  + ``eventhours``:
+    Time interval in hours to generate a record.
+
+
+Trigger
+=======
+
+Example:
+
+.. code-block:: xml
+
+   <trigger id="1" enable="true">
+           <name>t_pstore</name>
+           <type>node</type>
+           <path>/sys/fs/pstore/console-ramoops-0</path>
+   </trigger>
+
+* ``name``:
+  The name of trigger. It's used by crash and info configuration module.
+* ``type`` and ``path``:
+  These two labels are used to get the content of trigger files.
+  ``Types`` have been implemented:
```

Comment:
I'm not sure what you mean by ``types`` here?  I don't see something that looks like a list of types?  Are the types "node", "file", "dir",  "rebootreason", and "cmd"?  If it is, this should be stated more clearly, and describe what the different types mean.

Reply:
OK, I will refine this part.